python/test/unit/cuda/test_experimental_tma.py

```diff
@@ -1177,3 +1177,275 @@ def alloc_fn(size: int, align: int, steam):
+    def kernel(Y, X, N, BLOCK_SIZE: tl.constexpr):
+        pid = tl.program_id(0)
+        offs = pid * BLOCK_SIZE + tl.arange(0, BLOCK_SIZE)
+        mask = offs < N
+        x = tl.load(X + offs, mask=mask)
+        tl.store(Y + offs, x, mask=mask)
+
+    ret = torch.empty(x.shape, dtype=torch.float16, device=x.device)
+    grid = lambda META: (triton.cdiv(x.numel(), META['BLOCK_SIZE']), )
+    dtype = getattr(tl, dtype)
+    kernel[grid](ret, triton.reinterpret(x, dtype), ret.numel(), BLOCK_SIZE=1024)
+    return ret
+
+
+@triton.jit
+def mxfp8_mxfp4_matmul_tma(  #
+        a_ptr, b_ptr, output_ptr,  #
+        a_scale, b_scale,  #
+        M, N, K,  #
+        stride_scale,  #
+        stride_am, stride_ak,  #
+        stride_cm, stride_cn,  #
+        BLOCK_M: tl.constexpr,  #
+        BLOCK_N: tl.constexpr,  #
+        BLOCK_K: tl.constexpr,  #
+        NUM_STAGES: tl.constexpr):  #
+    pid = tl.program_id(axis=0)
+    num_pid_m = tl.cdiv(M, BLOCK_M)
+    pid_m = pid % num_pid_m
+    pid_n = pid // num_pid_m
+    offs_am = (pid_m * BLOCK_M + tl.arange(0, BLOCK_M)) % M
+    offs_bn = (pid_n * BLOCK_N + tl.arange(0, BLOCK_N)) % N
+    offs_bn_tma = pid_n * BLOCK_N
+    offs_ak = tl.arange(0, BLOCK_K)
+    offs_scale_k = tl.arange(0, BLOCK_K // 32)
+    a_scale_ptr = a_scale + offs_am[:, None] * stride_scale + offs_scale_k[None, :]
+    b_scale_ptr = b_scale + offs_bn[:, None] * stride_scale + offs_scale_k[None, :]
+    a_ptrs = a_ptr + (offs_am[:, None] * stride_am + offs_ak[None, :] * stride_ak)
+    accumulator = tl.zeros((BLOCK_M, BLOCK_N), dtype=output_ptr.dtype.element_ty)
+    offs_bk = 0
+
+    b_desc = tl._experimental_make_tensor_descriptor(
+        b_ptr,
+        shape=[N, K],
+        strides=[K // 2, 1],
+        block_shape=[BLOCK_N, BLOCK_K // 2],
+    )
+
+    for k in tl.range(0, tl.cdiv(K, BLOCK_K), num_stages=NUM_STAGES):
+        a = tl.load(a_ptrs)
+        b = b_desc.load([offs_bn_tma, offs_bk])
+
+        scale_a = tl.load(a_scale_ptr)
+        scale_b = tl.load(b_scale_ptr)
+        accumulator = tl.dot_scaled(a, scale_a, "e5m2", b.T, scale_b, "e2m1", accumulator)
+        a_ptrs += BLOCK_K * stride_ak
+
+        offs_bk += BLOCK_K
+        a_scale_ptr += BLOCK_K // 32
+        b_scale_ptr += BLOCK_K // 32
+
+    offs_cm = pid_m * BLOCK_M + tl.arange(0, BLOCK_M)
+    offs_cn = pid_n * BLOCK_N + tl.arange(0, BLOCK_N)
+    output_ptrs = output_ptr + stride_cm * offs_cm[:, None] + stride_cn * offs_cn[None, :]
+    c_mask = (offs_cm[:, None] < M) & (offs_cn[None, :] < N)
+    tl.store(output_ptrs, accumulator, mask=c_mask)
+
+
+@requires_tma
+@pytest.mark.parametrize("M, N, K", [(1024, 512, 256), (128, 256, 256)])
+@pytest.mark.parametrize("BLOCK_M, BLOCK_N, BLOCK_K", [(128, 128, 256), (128, 256, 256)])
+# TODO: BLOCK_K == 128
+#@pytest.mark.parametrize("BLOCK_M, BLOCK_N, BLOCK_K", [(128, 128, 128), (128, 128, 256), (128, 256, 128),
+#                                                       (128, 256, 256)])
```

Comment:
Is this TODO related to the discussion in https://github.com/triton-lang/triton/pull/6045/files#r1990716965? At first glance I don't understand why `BLOCK_K = 128` would not work

Reply:
Yes, this test raises illegal instruction error for `BLOCK_K=128` when run under `cuda-gdb`. Did you try it with your PR to see if it has the same issue?

Reply:
I just tried running the test with `BLOCK_K = 128` under cuda-gdb or `CUDA_LAUNCH_BLOCKING=1`, no error occurred. 

Due to padding in SMEM, `BLOCK_K = 128` implies `contig_dim_size` is exactly 128. However, you said `BLOCK_K = 128` leads to "smaller than 128" `contig_dim_size` - maybe you are using `getShapePerCTA` when you should be using `getAllocShapePerCTA`. The former returns logical shape which for `fp4Padded` returns the packed shape.